front/lib/api/assistant/configuration.ts

```diff
@@ -740,6 +747,12 @@ export async function createAgentConfiguration(
   let version = 0;
   let listStatusOverride: AgentUserListStatus | null = null;
 
+  const groupIds = (
+    await DataSourceViewResource.fetchByIds(auth, dataSourceViewIds)
+  )
+    .map((view) => view.acl().aclEntries.map((entry) => entry.groupId))
```

Comment:
Let's not hack it 😛. You can add a groupIds method on the `VaultResource`.

Reply:
wouldn't call it a hack. creating a new word in resource each time we need anything seems a bit overkill no?
better to do 100 things by composing use of 10 functions in 10 different ways rather than by creating 10 functions

Reply:
(if we needed that 2 or 3 places elsewhere I would agree, but for single uses IMHO we should not create a new function)

Reply:
It's just that it seems pretty complex just to get the group ids of a data source, right 😛?